File: backend/app/schemas/incident.py
from pydantic import BaseModel
from datetime import datetime
from app.models.core import IncidentStatus

class IncidentCreate(BaseModel):
    title: str
    description: str
    category: str
    severity: int | None = None
    latitude: float
    longitude: float
    address: str | None = None

class IncidentResponse(BaseModel):
    id: int
    title: str
    description: str
    category: str
    severity: int | None = None
    report_count: int
    latitude: float
    longitude: float
    address: str | None = None
    status: IncidentStatus
    trust_status: str
    reported_by: int
    created_at: datetime
    
    # NEW FIELDS ADDED HERE
    upvotes: int
    is_verified: bool

    class Config:
        from_attributes = True


class IncidentReportDetail(BaseModel):
    id: int
    user_id: int
    user_name: str
    user_email: str
    created_at: datetime
    latitude: float
    longitude: float
    image_filename: str | None = None


class IncidentReportHistoryResponse(BaseModel):
    incident_id: int
    incident_title: str
    reports: list[IncidentReportDetail]


class PendingResolutionRequestResponse(BaseModel):
    request_id: int
    incident_id: int
    incident_title: str
    category: str
    requested_at: datetime
    requested_by_admin_name: str
    requested_by_admin_email: str


class ResolutionRequestAction(BaseModel):
    resolved: bool
    message: str | None = None


class ResolutionRequestActionResponse(BaseModel):
    request_id: int
    incident_id: int
    status: str
    response_message: str | None = None


class AdminResolutionFeedbackResponse(BaseModel):
    request_id: int
    incident_id: int
    incident_title: str
    category: str
    user_name: str
    user_email: str
    status: str
    response_message: str | None = None
    created_at: datetime
    responded_at: datetime | None = None

File: mobile/lib/models/incident.dart
class Incident {
  final int id;
  final String title;
  final String description;
  final String category;
  final int? severity;
  final double latitude;
  final double longitude;
  final String? address;
  final String? image;
  final String status;
  final String trustStatus;
  final int reportCount;
  final int upvotes;
  final bool isVerified;
  final DateTime createdAt;

  Incident({
    required this.id,
    required this.title,
    required this.description,
    required this.category,
    this.severity,
    this.image,
    required this.latitude,
    required this.longitude,
    this.address,
    required this.status,
    required this.trustStatus,
    required this.reportCount,
    required this.upvotes,
    required this.isVerified,
    required this.createdAt,
  });

  factory Incident.fromJson(Map<String, dynamic> json) {
    return Incident(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? 'Other',
      severity: json['severity'],
      image: json['image'],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      address: json['address'],
      status: json['status'] ?? 'active',
      trustStatus: json['trust_status'] ?? 'Unknown',
      reportCount: json['report_count'] ?? 1,
      upvotes: json['upvotes'] ?? 0,
      isVerified: json['is_verified'] ?? false,
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
    );
  }
}
